dsa-solutions/lc-solutions/0100-0199/0152-Maximum-Product-Subarray.md

```diff
@@ -0,0 +1,161 @@
+---
+id: maximum-product-subarray
+title: Maximum Product Subarray
+sidebar_label: 0152 Maximum Product Subarray
+tags:
+  - Array
+  - Dynamic Programming
+  - LeetCode
+  - Python
+  - Java
+  - C++
+description: "This is a solution to the Maximum Product Subarray problem on LeetCode."
+---
+
+## Problem Description
+
+Given an integer array nums, find a subarray that has the largest product, and return the product.
+
+The test cases are generated so that the answer will fit in a 32-bit integer.
+
+### Examples
+
+**Example 1:**
+
+```
+Input: nums = [2,3,-2,4]
+Output: 6
+Explanation: [2,3] has the largest product 6.
+```
+
+**Example 2:**
+
+```
+Input: nums = [-2,0,-1]
+Output: 0
+Explanation: The result cannot be 2, because [-2,-1] is not a subarray.
+```
+
+### Constraints
+
+- $1 <= nums.length <= 2 * 10**4$
+- $10 <= nums[i] <= 10$
+- $The product of any prefix or suffix of nums is guaranteed to fit in a 32-bit integer.$
```

Comment:
replace `$The product of any prefix or suffix of nums is guaranteed to fit in a 32-bit integer.$` to `The product of any prefix or suffix of nums is guaranteed to fit in a 32-bit integer.`